File: CS/SilverlightApplication1/MainPageViewModel.cs
using System.ComponentModel;
using System.Windows.Input;
using DevExpress.Mvvm;
using DevExpress.Xpf.Printing;
// ...

namespace SilverlightApplication1 {
    public class MainPageViewModel {
        public IDocumentPreviewModel PreviewModel { get; private set; }
        public ICommand CreateDocumentCommand { get; private set; }
        public DelegateCommand ClearDocumentCommand { get; private set; }

        public MainPageViewModel() {
            PreviewModel = new ReportServicePreviewModel {
                ServiceUri = "../ReportService1.svc",
                ReportName = "SilverlightApplication1.Web.XtraReport1, SilverlightApplication1.Web"
            };
            ClearDocumentCommand = new DelegateCommand(ExecuteClearDocumentCommand, CanExecuteClearDocumentCommand);
            CreateDocumentCommand = new DelegateCommand(ExecuteCreateDocumentCommand);
        }

        void ExecuteCreateDocumentCommand() {
            PreviewModel.CreateDocument();
            ClearDocumentCommand.RaiseCanExecuteChanged();
        }

        bool CanExecuteClearDocumentCommand() {
            return !PreviewModel.IsEmptyDocument;
        }

        void ExecuteClearDocumentCommand() {
            PreviewModel.Clear();
            ClearDocumentCommand.RaiseCanExecuteChanged();
        }

        #region INotifyPropertyChanged

        public event PropertyChangedEventHandler PropertyChanged;

        void RaisePropertyChanged(string propertyName) {
            if(PropertyChanged != null) {
                PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
            }
        }
        #endregion


    }
}

File: VB/SilverlightApplication1/MainPageViewModel.vb
Imports Microsoft.VisualBasic
Imports System.ComponentModel
Imports System.Windows.Input
Imports DevExpress.Mvvm
Imports DevExpress.Xpf.Printing
' ...

Namespace SilverlightApplication1
	Public Class MainPageViewModel
		Private privatePreviewModel As IDocumentPreviewModel
		Public Property PreviewModel() As IDocumentPreviewModel
			Get
				Return privatePreviewModel
			End Get
			Private Set(ByVal value As IDocumentPreviewModel)
				privatePreviewModel = value
			End Set
		End Property
		Private privateCreateDocumentCommand As ICommand
		Public Property CreateDocumentCommand() As ICommand
			Get
				Return privateCreateDocumentCommand
			End Get
			Private Set(ByVal value As ICommand)
				privateCreateDocumentCommand = value
			End Set
		End Property
		Private privateClearDocumentCommand As DelegateCommand
		Public Property ClearDocumentCommand() As DelegateCommand
			Get
				Return privateClearDocumentCommand
			End Get
			Private Set(ByVal value As DelegateCommand)
				privateClearDocumentCommand = value
			End Set
		End Property

		Public Sub New()
			PreviewModel = New ReportServicePreviewModel With {.ServiceUri = "../ReportService1.svc", .ReportName = "SilverlightApplication1.Web.XtraReport1, SilverlightApplication1.Web"}
			ClearDocumentCommand = New DelegateCommand(AddressOf ExecuteClearDocumentCommand, AddressOf CanExecuteClearDocumentCommand)
			CreateDocumentCommand = New DelegateCommand(AddressOf ExecuteCreateDocumentCommand)
		End Sub

		Private Sub ExecuteCreateDocumentCommand()
			PreviewModel.CreateDocument()
			ClearDocumentCommand.RaiseCanExecuteChanged()
		End Sub

		Private Function CanExecuteClearDocumentCommand() As Boolean
			Return Not PreviewModel.IsEmptyDocument
		End Function

		Private Sub ExecuteClearDocumentCommand()
			PreviewModel.Clear()
			ClearDocumentCommand.RaiseCanExecuteChanged()
		End Sub

		#Region "INotifyPropertyChanged"

		Public Event PropertyChanged As PropertyChangedEventHandler

		Private Sub RaisePropertyChanged(ByVal propertyName As String)
			RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
		End Sub
		#End Region


	End Class
End Namespace
